frontend/src/js/small-tab-navigation/HoverNavigatable.tsx

```diff
@@ -0,0 +1,41 @@
+import { FC } from "react";
+import { useDrop } from "react-dnd";
+import { DNDType } from "../common/constants/dndTypes";
+
+
+interface PropsT {
+    timeUntilNavigateOverride?: number;
+    triggerNavigate: () => void;
+    disabled?: boolean;
+}
+
+// default time until the hover triggers navigation
+// 600 ms is the default time for a tooltip to appear
+// feels very responsive
+// can be overriden by passing a timeUntilNavigateOverride prop
+const TIME_UNTIL_NAVIGATE = 600;
+
+export const HoverNagiatable:FC<PropsT> = ({ timeUntilNavigateOverride, triggerNavigate, children, disabled }) => {
+    const time_to_navigate = timeUntilNavigateOverride || TIME_UNTIL_NAVIGATE;
+    let timeout: number | null | NodeJS.Timeout; // timeout can either be a number or a NodeJS.Timeout depending on the environment
```

Comment:
What happens when the component rerenders? Will the timeout just get overridden? I think we should probably use a timeout ref instead of a local variable